types/releases.ts

```diff
@@ -0,0 +1,37 @@
+// TODO(HinataKah0): Remove this, see https://github.com/nodejs/nodejs.org/issues/5303#issuecomment-1518947513
+/* eslint-disable no-unused-vars */
+
+export enum ReleaseTypes {
+  current = 'Current',
+  lts = 'LTS',
+  maintenance = 'Maintenance',
+  endoflife = 'End-of-life',
+}
+
+export interface UpcomingReleaseData {
+  releaseDate: string;
+  releaseType: ReleaseTypes;
+  alreadyReleased: boolean;
+}
+
+export interface UpcomingRelease {
```

Comment:
I just realized this.
Nit: I am thinking of renaming this to be `UpcomingReleases`.
What do you think?

Reply:
`UpcomingReleases` sounds fine to me if you wish to change it

Reply:
Where was this `UpcomingRelease` being used? Because it has a title and releases type?

Reply:
It is used in the utility function [`getUpcomingReleases`](https://github.com/nodejs/nodejs.dev/blob/main/src/util/getUpcomingReleases.ts#L9-L12) and the component [`UpcomingReleases`](https://github.com/nodejs/nodejs.dev/blob/main/src/components/DownloadComponents/UpcomingReleases/index.tsx#L7-L9) alongside its' parent [`DownloadReleases`](https://github.com/nodejs/nodejs.dev/blob/main/src/components/DownloadComponents/DownloadReleases/index.tsx#L8-L11) (both usages are Prop's type).

I just realized it conflicts with the component name if we name it to be `UpcomingReleases` 😅 
An alternative will be `UpcomingReleasesData` so it's consistent with others within the same file

Reply:
I mean why the type has an array of releases and a title as its type definition.

I'm more questioning the current composition of the type. Maybe we don't even need this type at all or we can change how the components are when we migrate them.

Feels kinda weird an "UpcomingRelease" with a title and multiple releases on it. Even if it is "UpcomingReleases" (what is the meaning of the title type?)

Reply:
Ah, sorry I misunderstood you.

![Screenshot 2023-04-28 at 5 50 31 PM](https://user-images.githubusercontent.com/128208841/235116643-980c7f41-2bdb-4f2c-8b51-dc5d663233f6.png)

`title` is actually a bit misleading :)
It is actually the major release version. So, `UpcomingRelease` basically groups all the releases by their major version.

`UpcomingMajorVersionReleases` or something along that line.

Reply:
Can we rename this to `upcomingRelease.name` then?

Reply:
I think it's better to be more explicit `upcomingRelease.majorVersion`?

Reply:
It's kinda implicit that the `name` is the name of a release. (major)

Kinda verbose naming it `majorVersion` (IMHO)

Reply:
> name of a release

Hmm, yeah, it actually makes sense.
+1 for `name`

Reply:
I don't want to derail this PR, but IMO the whole https://nodejs.dev/en/download/ page is a worse UX than the existing https://nodejs.org/en/download. Getting to the actual downloads (if you need something other than the three prominent buttons at the top of the page) requires more clicks and is after a section that overlaps with https://nodejs.dev/en/about/releases/.

Reply:
@richardlau I'd say that's a topic for a Discussion (Discussions Tab). I definitely agree that the new redesigned Downloads page could show other platforms like we do on nodejs.org, but maybe on the bottom.

That's something to design/spec-out but not on this PR 😅

Reply:
> Getting to the actual downloads (if you need something other than the three prominent buttons at the top of the page) requires more clicks and is after a section that overlaps with

Yeah that page should be removed tbh. But I disagree that it requires more clicks. You just need to scroll down to see more platforms at the bottom. But we should add even more platforms there and useful links.